build.gradle.kts

```diff
@@ -0,0 +1,226 @@
+            client()
+            ideConfigGenerated(project.rootProject == project)
+            name("Networking Testmod (Client)")
+            source(sourceSets.getByName("testmod"))
+        }
+    }
+}
+
+configurations.implementation.get().extendsFrom(configurations.shadow.get())
+
+/*
+// Doesn't work, idk why
+val modImplementationAndInclude: Configuration by configurations.creating {
+    extendsFrom(configurations["implementation"])
+    extendsFrom(configurations["include"])
+}
+ */
+
+repositories {
+    maven("https://maven.fabricmc.net/")
+    maven("https://maven.nucleoid.xyz/")
+    maven("https://maven.bymartrixx.me/")
+    maven("https://jitpack.io")
+    maven("https://oss.sonatype.org/content/repositories/snapshots")
+    mavenCentral()
+    jcenter()
+}
+
+dependencies {
+    // To change the versions see the libs.versions.toml
+
+    // Fabric
+    minecraft(libs.minecraft)
+    mappings(variantOf(libs.yarn.mappings) { classifier("v2") })
+    modImplementation(libs.fabric.loader)
+
+    // Fabric API
+    modImplementation(libs.fabric.api)
+
+    // Permissions
+    modImplementation(libs.fabric.permissions)
+    include(libs.fabric.permissions)
+
+    // Translations
+    modImplementation(libs.translations)
+    include(libs.translations)
+
+    // Kotlin
+    modImplementation(libs.fabric.kotlin)
+
+    // Database
+    shadow(libs.exposed.core)
+    shadow(libs.exposed.dao)
+    shadow(libs.exposed.jdbc)
+    shadow(libs.exposed.java.time)
+    shadow(libs.sqlite.jdbc)
+
+    // Config
+    shadow(libs.konf.core)
+    shadow(libs.konf.toml)
+
+    // Debug
+    modRuntime(libs.wdmcf)
+}
+
+tasks {
+    val javaVersion = JavaVersion.VERSION_16
+
+    processResources {
+        inputs.property("id", modId)
+        inputs.property("name", modName)
+        inputs.property("version", modVersion)
+
+        filesMatching("fabric.mod.json") {
+            expand(
+                mapOf(
+                    "id" to modId,
+                    "version" to modVersion,
+                    "name" to modName,
+                    "fabricLoader" to libs.versions.fabric.loader.get(),
+                    "fabricApi" to libs.versions.fabric.api.get(),
+                    "fabricKotlin" to libs.versions.fabric.kotlin.get(),
+                )
+            )
+        }
+    }
+
+    withType<JavaCompile> {
+        options.encoding = "UTF-8"
+        sourceCompatibility = javaVersion.toString()
+        targetCompatibility = javaVersion.toString()
+        options.release.set(javaVersion.toString().toInt())
+    }
+
+    withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+        kotlinOptions { jvmTarget = javaVersion.toString() }
+        sourceCompatibility = javaVersion.toString()
+        targetCompatibility = javaVersion.toString()
+    }
+
+    jar {
+        from("LICENSE")
+    }
+
+    java {
+        toolchain { languageVersion.set(JavaLanguageVersion.of(javaVersion.toString())) }
```

Comment:
As the mixins file also has java version, we should either expose that for gradle to automatically handle or remove the java version option entirely, in favour of hardcoding